subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformationResult.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.transform;
+
+import com.google.common.collect.ImmutableList;
+
+import java.io.File;
+import java.util.stream.Stream;
+
+/**
+ * The result of running a transformation.
+ */
+public interface TransformationResult {
+    /**
+     * The result of the artifact transform, using relative paths into the provided input artifact.
+     */
+    ImmutableList<File> resultRelativeTo(File inputArtifact);
+
+    /**
+     * The result of the artifact transform in the transform workspace.
```

Comment:
```suggestion
     * The part of the result of the artifact transform that was produced in the transform workspace.
     Does not include results selected from the inputs.
```

Also, I think the term "result" is a bit confusing here, and using "outputs" would be better.

Reply:
I changed the names in the class like we said. PTAL.